Dockerfile

```diff
@@ -2,12 +2,7 @@ ARG BUILD_TYPE=dev
 ARG BUILDER_BASE=quay.io/confidential-containers/golang-fedora:1.20.8-36
 ARG BASE=registry.fedoraproject.org/fedora:38
 
-# This dockerfile uses Go cross-compilation to build the binary,
-# we build on the host platform ($BUILDPLATFORM) and then copy the
-# binary into the container image of the target platform ($TARGETPLATFORM)
-# that was specified with --platform. For more details see:
-# https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/
-FROM --platform=$BUILDPLATFORM $BUILDER_BASE as builder-release
+FROM --platform=$TARGETPLATFORM $BUILDER_BASE as builder-release
```

Comment:
What is the reason for this change? This will make builds really slow, as it does not use Go cross compilation but qemu emulation of the builder.

Reply:
@katexochen this is to enable multi-arch dev-* build to include libvirt options. Maybe we can split it to dev-* and release here.

Reply:
You just need to set `supported_arches` in `--platform "${supported_arches}"` in the docker call to the platform you want to build.

Reply:
@katexochen when `CGO=1` set (which is the case when libvirt enabled), I think we need use `TARGETPLATFORM` builder to build the right image. I did some trying in this PR https://github.com/confidential-containers/cloud-api-adaptor/pull/1439 before.